task_processing/interfaces/task_executor.py

```diff
@@ -24,14 +21,14 @@ class TaskExecutor(object):
     """
 
     @abc.abstractmethod
-    def run(self, task_config):
-        """Run the supplied task
+    def run(self, task_config, task_id):
```

Comment:
hmm... I don't understand why we need to pass task_id separately? Also, why are we using a vector?